### PR TITLE
Adding Storagepool tier datasource

### PR DESCRIPTION
**Pull request by ABHISHEK-SINHA10**
# Description
A few sentences describing the overall goals of the pull request's commits.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# ISSUE TYPE

- Feature Pull Request

##### OUTPUT

Image

##### ADDITIONAL INFORMATION

```paste below

```
# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by terraform-providers-svc** [comment]
Can one of the admins verify this patch?

File: examples/data-sources/powerscale_storagepool_tier/data-source.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/*
+Copyright (c) 2023-2024 Dell Inc., or its subsidiaries. All Rights Reserved.
```
Comment (shenda1) [inline examples/data-sources/powerscale_storagepool_tier/data-source.tf:2]: 2024

File: powerscale/provider/storagepool_tier_datasource.go [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+			"The information fetched from this datasource can be used for getting the details or for further processing in resource block. " +
+			"PowerScale Storagepool tiers provide the ability to configure Storagepool tiers on the cluster.",
+		Description: "This datasource is used to query the Storagepool tiers from PowerScale array. " +
+			"The information fetched from this datasource can be used for getting the details or for further processing in resource block. " +
+			"PowerScale Storagepool tiers provide the ability to configure Storagepool tiers on the cluster.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "Id of Storagepool tiers. Readonly. Fixed value of \"storagepool_tiers\"",
+				MarkdownDescription: "Id of Storagepool tiers. Readonly. Fixed value of \"storagepool_tiers\"",
+				Optional:            false,
```
Comment (shenda1) [inline powerscale/provider/storagepool_tier_datasource.go:67]: No need to mention attributes if its false

File: powerscale/provider/storagepool_tier_datasource.go [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+				Required:            false,
+				Computed:            true,
+			}, // Need to created nested attributes
```
Comment (shenda1) [inline powerscale/provider/storagepool_tier_datasource.go:70]: Remove this comment if not required

File: examples/data-sources/powerscale_storagepool_tier/data-source.tf (hunk 1 above)
Comment (ABHISHEK-SINHA10) [reply examples/data-sources/powerscale_storagepool_tier/data-source.tf]: Done

File: powerscale/provider/storagepool_tier_datasource.go (hunk 2 above)
Comment (ABHISHEK-SINHA10) [reply powerscale/provider/storagepool_tier_datasource.go]: removed

File: powerscale/provider/storagepool_tier_datasource.go (hunk 3 above)
Comment (ABHISHEK-SINHA10) [reply powerscale/provider/storagepool_tier_datasource.go]: removed

**Comment by ABHISHEK-SINHA10** [comment]
UT
(coverage)
Acceptance testing
Mock server calls

**Comment by shenda1** [comment]
run e2e test StoragepoolTierDatasource

File: powerscale/provider/storagepool_tier_datasource.go [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+			"The information fetched from this datasource can be used for getting the details or for further processing in resource block. " +
+			"PowerScale Storagepool tiers provide the ability to configure Storagepool tiers on the cluster.",
+		Description: "This datasource is used to query the Storagepool tiers from PowerScale array. " +
+			"The information fetched from this datasource can be used for getting the details or for further processing in resource block. " +
+			"PowerScale Storagepool tiers provide the ability to configure Storagepool tiers on the cluster.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "Id of Storagepool tiers. Readonly. Fixed value of \"storagepool_tiers\"",
+				MarkdownDescription: "Id of Storagepool tiers. Readonly. Fixed value of \"storagepool_tiers\"",
+				Required:            false,
```
Comment (shenda1) [inline powerscale/provider/storagepool_tier_datasource.go:67]: false is implicit. No need to specify explicitly.

**Comment by shenda1** [comment]
run e2e test StoragepoolTierDatasource

**Review by Krunal-Thakkar — APPROVED** [review]

File: powerscale/provider/storagepool_tier_datasource_test.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+}
+`
+
+var StoragepoolTierDatasourceAllWithResourceConfig = `
+resource "powerscale_storagepool_tier" "example" {
+	name = "TestExample"
+}
+
+data "powerscale_storagepool_tier" "all_test" {
+
```
Comment (shenda1) [inline powerscale/provider/storagepool_tier_datasource_test.go:93]: Use depends_on to add the dependency

File: powerscale/provider/storagepool_tier_datasource.go [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+							ElementType:         types.StringType,
+						},
+						"lnns": schema.ListAttribute{
+							Description:         "Apply to files with manually-managed I/O optimization settings.",
+							MarkdownDescription: "Apply to files with manually-managed I/O optimization settings.",
+							Computed:            true,
+							ElementType:         types.Int32Type,
+						},
+						"name": schema.StringAttribute{
+							Description:         "Manage protection settings.",
```
Comment (rounak-adhikary) [inline powerscale/provider/storagepool_tier_datasource.go:93]: Nit: Can you please double-check and confirm that these descriptions are correct?

**Review by shenda1 — APPROVED** [review]

**Comment by shenda1** [comment]
run e2e test StoragepoolTierDatasource

File: powerscale/provider/storagepool_tier_datasource.go (hunk 6 above)
Comment (rounak-adhikary) [reply powerscale/provider/storagepool_tier_datasource.go]: Hi @ABHISHEK-SINHA10 can you pls confirm the descriptions for all the other fields too? 
For example, I see that the description of `children` is `Manage I/O optimization settings`.

Comment (ABHISHEK-SINHA10) [reply powerscale/provider/storagepool_tier_datasource.go]: @rounak-adhikary  Thanks for pointing out, I have checked all the description and now it's updated, please check once again

**Comment by ABHISHEK-SINHA10** [comment]
image
image


Testcase working fine with UT and mock server test(No changes needed on mock server from my side, it's working fine after Trisha's PR on mock server)

**Comment by shenda1** [comment]
run e2e test StoragepoolTierDatasource

**Pull request merged**